tests/math/test_matrix_manipulation.py

```diff
@@ -1188,9 +1188,33 @@ def test_sqrt_matrix_sparse_input_valid(self):
 CONVERGENCE_ERROR = "Convergence threshold not reached"
 
 
+def _reverse_det_hermitian(mat):
+    """Helper function to reverse determinant sign of Hermitian matrix"""
+    eigvals, eigvecs = np.linalg.eigh(mat)
+    eigvals[0] *= -1
+    return eigvecs @ np.diag(eigvals) @ eigvecs.T.conj()
+
+
+def _enforce_positivity(mat):
+    """Helper function to ensure matrix is positive definite"""
+    det_mat = np.linalg.det(mat)
+    return mat if det_mat > 0 else _reverse_det_hermitian(mat)
+
+
 class TestDenmanBeaversIterations:
     """Tests for the Denman-Beavers iteration method for matrix square root"""
 
+    @staticmethod
+    def _reverse_det_hermitian(mat):
+        eigvals, eigvecs = np.linalg.eigh(mat)
+        eigvals[0] *= -1
+        return eigvecs @ np.diag(eigvals) @ eigvecs.T.conj()
+
+    @staticmethod
+    def _enforce_positivity(mat):
+        det_mat = np.linalg.det(mat)
+        return mat if det_mat > 0 else _reverse_det_hermitian(mat)
```

Comment:
If these are defined above, do we also need them as static methods here?
Which should we keep?

Reply:
Oh sorry forgot to delte the staticmethod decorator from my last try